packages/core/src/sdk/analytics/hooks/usePageViewEvent.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import { sendAnalyticsEvent } from '@faststore/sdk'
+import type { PageViewEvent } from '@faststore/sdk'
+
+export const usePageViewEvent = () => {
```

Comment:
The code looks fine. Just a question, **Page view** event is not an event that is already generated by default when adding an analytics tool like G.A?

Reply:
https://developers.google.com/analytics/devguides/collection/ga4/views?hl=pt-br&client_type=gtag

Reply:
What is the motivation for sending this manually?

Reply:
We were checking this again, and it is not automatically firing the event. I attached a [video](https://www.loom.com/share/7175880139e145fd9eaaa2f978717923) using Tag Assistant to debug both in their current store and in the PR with the proposed changes.

You will see that the tags are only firing when we actually add the event. We assume that this is related to how NextJS handle events, as it is a history change and not a route change. NextJS leverages the SPA features to speed up the navigation by replacing the regular anchor (<a>) with the [Link](https://nextjs.org/docs/pages/api-reference/components/link) one. So, it's necessary to add that page_view event call to make sure that nextjs functionally + GA4 works as expected.

Reply:
It makes sense @higaeduardo . Apparently Next is using [history push state](https://www.google.com/search?q=HTML5+pushstate+API) and this is not triggering the **page_view** event. There is a specific event for [changing history](https://support.google.com/tagmanager/answer/7679322?hl=en) in Google Tag Manager that can be associated with a tag. But since all stores will use **next js** from now on, I think it makes sense to have this logic of triggering the page_view event within our Framework.

I also find this good article about it:  https://caisy.io/blog/gtm-nextjs-guide

Next steps: